assets/js/state/selectors/sapSystems.js

```diff
@@ -0,0 +1,118 @@
+import { filter } from 'lodash';
+import { createSelector } from '@reduxjs/toolkit';
+
+import { APPLICATION_TYPE, DATABASE_TYPE } from '@lib/model';
+import { getCluster } from '@state/selectors/cluster';
+import { getHost } from '@state/selectors/host';
+
+const enrichInstance = (instance) => (state) => {
+  const host = getHost(instance.host_id)(state);
+  const cluster = getCluster(host?.cluster_id)(state);
+
+  return {
+    ...instance,
+    host: {
+      ...host,
+      cluster,
+    },
+  };
+};
+
+const enrichInstances = (instances) => (state) =>
+  instances.map((instance) => enrichInstance(instance)(state));
+
+export const getEnrichedApplicationInstances = createSelector(
+  [
+    (state) =>
+      enrichInstances(state.sapSystemsList.applicationInstances)(state),
+  ],
+  (enrichedInstances) => enrichedInstances
+);
+
+export const getEnrichedDatabaseInstances = createSelector(
+  [(state) => enrichInstances(state.sapSystemsList.databaseInstances)(state)],
+  (enrichedInstances) => enrichedInstances
+);
+
+export const getSapSystem = (id) => (state) =>
+  state.sapSystemsList.sapSystems.find((sapSystem) => id === sapSystem.id);
+
+export const getDatabase = (id) => (state) =>
+  state.databasesList.databases.find((database) => id === database.id);
+
+export const getEnrichedSapSystemDetails = createSelector(
+  [
+    (id, state) => getSapSystem(id)(state),
+    (id, state) =>
+      enrichInstances(
```

Comment:
I'm not sure if adding this operation here is a good idea, or it should better go in the body of the function